azure-common/azure/common/client_factory.py

```diff
@@ -7,11 +7,22 @@
 import io
 import os
 import adal
+import inspect
+import json
 from msrestazure.azure_active_directory import AdalAuthentication
 
 from .credentials import get_azure_cli_credentials, get_cli_profile
 from .cloud import get_cli_active_cloud
 
+def _instanciate_client(clientclass, **kwargs):
+    """Instanciate client from kwargs, remove parameters not needed by the Client.
+    """
+    args = inspect.getargspec(clientclass.__init__).args
```

Comment:
Be aware that `inspect.getargspec()` is deprecated since Python 3.0. You want [`inspect.getfullargspec()`](https://docs.python.org/3/library/inspect.html#inspect.getargspec) on Python 3.